Lib/fontParts/base/font.py

```diff
@@ -353,11 +366,27 @@ def generate(self, format, path=None):
             fileName += ext
             path = os.path.join(path, fileName)
         path = normalizers.normalizeFilePath(path)
-        return self._generate(format=format, path=path)
+        return self._generate(
+            format=format,
+            path=path,
+            environmentOptions=environmentOptions
+        )
 
     generate.__doc__ %= generateFormatToExtension.__doc__
 
-    def _generate(self, format, path, **kwargs):
+    def _isValidGenerateEnvironmentOption(self, name):
```

Comment:
Codacy complains that this could be a `@staticmethod` as `self` isn't used. I don't think that an environment will need `self` for this, so perhaps we should decorate it with `@staticmethod`. I'm ok either ways.